client/lib/src/api_client.dart

```diff
@@ -32,7 +32,7 @@ class ApiClient {
   WebSocket _webSocket;
 
   void init(String gameId, String player) {
-    final protocol = window.location.protocol == 'https' ? 'wss' : 'ws';
+    final protocol = window.location.protocol.contains('https') ? 'wss' : 'ws';
     _webSocket = WebSocket('${protocol}://${apiHost}/game/${gameId}/${player}');
```

Comment:
can you also change apiHost to be window.location.hostname?

you can remove the corresponding entry in build.yaml also